### ISSUE TITLE
[Task] Update Merlin Schema to include inputs, outputs, other metadata

### ISSUE THREAD

**Issue by EvenOldridge** [NVIDIA-Merlin/Merlin#640]
Metadata schema updated to include:
- [ ] Input Schema
- [ ] Output Schema
- [ ] Data distributions?
- [ ] Prediction distributions?

**Comment by nv-alaiacano** [comment]
Meeting notes (not super complete): https://docs.google.com/document/d/1k_QJ063LH0YIJ_wPVzwjFzTQXek0UtXXtmErj_kqrOY/edit#

**Comment by karlhigley** [comment]
I don't think this task is about changing the existing schema format; it's more about creating a format for capturing a broader set of metadata (including but not limited to schema information), right?
